src/treemodel/rootelement.js

```diff
@@ -19,16 +19,35 @@ export default class RootElement extends Element {
 	 *
 	 * @param {engine.treeModel.Document} doc {@link engine.treeModel.Document} that is an owner of the root.
 	 * @param {String} name Node name.
+	 * @param {String} rootName Root name inside parent {@link engine.treeModel.Document}.
 	 */
-	constructor( doc, name ) {
+	constructor( doc, name, rootName ) {
 		super( name );
 
 		/**
 		 * {@link engine.treeModel.Document} that is an owner of this root.
 		 *
 		 * @readonly
-		 * @member {engine.treeModel.Document} engine.treeModel.RootElement#doc
+		 * @member {engine.treeModel.Document} engine.treeModel.RootElement#document
 		 */
 		this.document = doc;
+
+		/**
+		 * Name of this root inside {@link engine.treeModel.Document} that is an owner of this root.
+		 *
+		 * @readonly
+		 * @member {String} engine.treeModel.RootElement#rootName
+		 */
+		this.rootName = rootName;
+	}
+
+	/**
+	 * Custom toJSON method to solve child-parent circular dependencies.
+	 *
+	 * @method engine.treeModel.RootElement#toJSON
+	 * @returns {String} Name of this root inside {@link engine.treeModel.Document} that is an owner of this root.
+	 */
+	toJSON() {
+		return typeof this.rootName === 'symbol' ? '$$graveyard' : this.rootName;
```

Comment:
Why you assume that every symbol is a `graveyard`? You could replace it with `this === this.document.graveyard`. But  I think that we should simplified it in general and use `$graveyard` string instead of using symbols. Also one `$` should be enough.